test/components/date-picker.spec.tsx

```diff
@@ -28,103 +37,102 @@ describe('The DatePicker Component', () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} />);
+        const {driver, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(select(datePickerInputId)).to.have.value(JANUARY_FIRST.toDateString()));
+        await waitForDom(() => expect(driver.input).to.have.value(JANUARY_FIRST.toDateString()));
     });
 
     it('should not call onChange with an invalid date', async () => {
         const onChange = sinon.spy();
-        const {select, waitForDom} = clientRenderer.render(<DatePicker onChange={onChange} />);
-        const datePickerInput = select(datePickerInputId);
+        const {driver} = clientRenderer.render(<DatePicker onChange={onChange} />)
+            .withDriver(DatePickerTestDriver);
 
-        trigger.change(datePickerInput!, '2sgsdfsdfw223');
-        simulate.blur(datePickerInput);
-
-        await waitForDom(() => expect(onChange).to.have.not.been.calledOnce);
+        driver.changeDate('2sgsdfsdfw223');
+        sleep(20);
+        expect(onChange).to.have.not.been.called;
     });
 
     it('should call onChange with the current input value when blurred', async () => {
         const onChange = sinon.spy();
-        const {select, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} onChange={onChange} />);
-        const datePickerInput = select(datePickerInputId);
+        const {driver, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} onChange={onChange} />)
+            .withDriver(DatePickerTestDriver);
 
-        trigger.change(datePickerInput!, '2017/02/01');
-        simulate.blur(datePickerInput);
+        driver.changeDate('2017/02/01');
 
         await waitForDom(() => expect(onChange).to.have.been.calledWithMatch({value: FEBRUARY_FIRST}));
     });
 
     it('should use a provided placeholder', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker placeholder="mm/dd/yyyy" />);
+        const {driver, waitForDom} = clientRenderer.render(<DatePicker placeholder="mm/dd/yyyy" />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(select(datePickerInputId)).to.have.attribute('placeholder', 'mm/dd/yyyy'));
+        await waitForDom(() => expect(driver.input).to.have.attribute('placeholder', 'mm/dd/yyyy'));
     });
 
     it('should show and hide the dropdown when the input is clicked', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker />);
-        const datePickerInput = select(datePickerInputId);
+        const {driver, waitForDom} = clientRenderer.render(<DatePicker />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
+        await waitForDom(() => expect(driver.dropDown).to.be.absent());
```

Comment:
consider adding `isOpen(): boolean` method to the driver. I assume users will like that more than asking whether the dropdown exists...